libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts

```diff
@@ -193,7 +219,29 @@ export class DelegationsIncomingService {
         'supportsProcuringHolders',
         'supportsCustomDelegation',
         'supportsPersonalRepresentatives',
+        'requireApiScopes',
       ],
     })
   }
+
+  private async getClientAllowedApiScopes(user: User): Promise<ApiScope[]> {
+    if (!user) return []
+
+    const clientAllowedScopes = (
+      await this.clientAllowedScopeModel.findAll({
+        where: {
+          clientId: user.client,
+        },
+      })
+    ).map((s) => s.scopeName)
+
+    return await this.apiScopeModel.findAll({
+      where: {
+        [Op.and]: [
+          { name: { [Op.in]: clientAllowedScopes.map((s) => s) } },
+          { enabled: true },
+        ],
+      },
+    })
```

Comment:
ApiScope is a pretty big model, would be nice to get only the attributes we need.

Though it's a bit risky since this array is being passed around between different services. Would be good to create an TypeScript interface for the object we pass around to catch unexpected attribute access.

Reply:
Fixed.